frontend/packages/process-editor/src/utils/processEditorUtils/processEditorUtils.ts

```diff
@@ -9,3 +11,26 @@ export const supportsProcessEditor = (version: string): boolean => {
   const firstNumber: number = Number(version[0]);
   return firstNumber > 7;
 };
+
+/**
+ * Updates the data task tracking lists
+ * Adds the provided item to the primary list and removes it from the secondary list (if it exists there)
+ * @param updatePrimaryList The function to update the primary list by adding the new item
+ * @param updateSecondaryList The function to update the secondary list by removing the item
+ * @param itemToAdd The item to add to the primary list
+ * @param secondaryList The secondary list to remove the item from
+ */
+export const updateDataTaskTrackingLists = (
+  updatePrimaryList: React.Dispatch<React.SetStateAction<BpmnDetails[]>>,
+  updateSecondaryList: React.Dispatch<React.SetStateAction<BpmnDetails[]>>,
+  itemToAdd: BpmnDetails,
+  secondaryList: BpmnDetails[],
+) => {
+  if (itemToAdd?.taskType !== 'data') {
+    return;
+  }
+  updatePrimaryList((prevItems: BpmnDetails[]) => [...prevItems, itemToAdd]);
+  if (secondaryList.some((task) => task.id === itemToAdd.id)) {
+    updateSecondaryList((prevItems) => prevItems.filter((task) => task.id !== itemToAdd.id));
+  }
```

Comment:
Since this method is already specific in its name, connecting it closely to handing data tasks, maybe we can specify the prop names a bit too so it is easier to understand what it does? 
updatePrimaryList --> addTaskToList
updateSecondaryList --> removeTaskFromList
itemToAdd --> taskToRemoveOrAddToProcess

Although I realize it is difficult to specify a method that already acts a bit general 😅 So maybe just leave it. Maybe we could add a scenario to the description or something

Reply:
I ended up removing this whole method. I tried extracting it to make it more testable, but I don't think it gives much value, as it just gets confusing what the context is. It's also not really possible to test the content of the function calls as you have commented above. So I have inlined it where it is used instead, hoping that will make it more readable 😊